### ISSUE TITLE
Override configuration options via Environment

### ISSUE THREAD

**Issue by mrueg** [cesanta/docker_auth#166]
It would be nice to be able to override config options via the environment as the docker registry currently does.

https://docs.docker.com/registry/configuration/#/override-specific-configuration-options

**Comment by rojer** [comment]
yes, sounds good to me. as always - PRs welcome.

**Comment by mrueg** [comment]
Seems like this is what's needed: https://github.com/docker/distribution/blob/master/configuration/configuration.go
https://github.com/docker/distribution/blob/master/configuration/parser.go

Do you think this will be difficult to integrate?

**Comment by rojer** [comment]
yes, i think taht since auth server is a companion to the registry, it should use the same configurations scheme. ideally - just directly reuse the one used by the registry.
however, [this](https://github.com/docker/distribution/blob/0700fa570d7bcc1b3e46ee127c4489fd25f4daa3/configuration/configuration.go#L17) seems like a blocker - we have fields with underscores already.
maybe something to consider for 2.0
